core/src/main/scala/kafka/controller/KafkaController.scala

```diff
@@ -1602,6 +1602,8 @@ class KafkaController(val config: KafkaConfig,
           processIsrChangeNotification()
         case Startup =>
           processStartup()
+        case ShutdownEventThread =>
+          // not handled here
```

Comment:
Shouldn't we throw an exception here too?

Reply:
Well, I have been still thinking about it; If we need an Exception, I will apply it in next cleanup. @abbccdda Could you tell me your opinion?

Reply:
Previously we would throw an exception (`MatchError`) and now we ignore it. That seems worse, right?

Reply:
Yes, you are right. I went ahead and merged because it was being addressed in #6686.

Reply:
Makes sense, we can just do it in #6686 then.